vmdb/app/models/ems_refresh/parsers/foreman.rb

```diff
@@ -0,0 +1,116 @@
+        result[:operating_system_flavors] = operating_system_flavors_inv_to_hashes(inv[:operating_systems], ids)
+        result
+      end
+
+      def configuration_inv_to_hashes(inv)
+        result = {}
+        ids = {}
+        result[:configuration_profiles] = configuration_profile_inv_to_hashes(inv[:hostgroups],
+                                                                              inv[:operating_system_flavors],
+                                                                              inv[:customization_scripts])
+        add_ids(ids, result[:configuration_profiles])
+        result[:configured_systems] = configured_system_inv_to_hashes(inv[:hosts], ids, inv[:operating_system_flavors])
+        result[:missing_key] = true if missing_key
+        result
+      end
+
+      def media_inv_to_hashes(media)
+        media.collect do |m|
+          {
+            "manager_ref" => "medium:#{m["id"]}",
+            "type"        => "CustomizationScriptMedium",
+            "name"        => m["name"]
+          }
+        end
+      end
+
+      def ptables_inv_to_hashes(ptables)
+        ptables.collect do |m|
+          {
+            "manager_ref" => "ptable:#{m["id"]}",
+            "type"        => "CustomizationScriptPtable",
+            "name"        => m["name"]
+          }
+        end
+      end
+
+      def customization_scripts_inv_to_hash(media, ptables)
+        media_inv_to_hashes(media) + ptables_inv_to_hashes(ptables)
+      end
+
+      def operating_system_flavors_inv_to_hashes(flavors_inv, ids)
+        flavors_inv.collect do |os|
+          {
+            "manager_ref"           => "operating_system:#{os["id"]}",
+            "name"                  => os["fullname"],
+            "description"           => os["description"],
+            "customization_scripts" => ids_lookup(ids, os["media"], "medium") + ids_lookup(ids, os["ptables"], "ptable")
+          }
+        end
+      end
+
+      def configuration_profile_inv_to_hashes(recs, osfs, scripts)
+        recs.collect do |profile|
+          {
+            "type"                           => "ConfigurationProfileForeman",
+            "manager_ref"                    => "hostgroup:#{profile["id"]}",
+            "name"                           => profile["name"],
+            "description"                    => profile["title"],
+            "operating_system_flavor_id"     => id_lookup(osfs, profile, "operating_system", "operatingsystem_id"),
+            "customization_script_ptable_id" => id_lookup(scripts, profile, "ptable"),
+            "customization_script_medium_id" => id_lookup(scripts, profile, "medium"),
+          }
+        end
+      end
+
+      def configured_system_inv_to_hashes(recs, profiles, operatingsystems)
+        recs.collect do |cs|
+          {
+            "type"                       => "ConfiguredSystemForeman",
+            "manager_ref"                => "host:#{cs["id"]}",
```

Comment:
That is the exception because we are merging two disparate things from the raw data.  Here it's just one thing.  Even so, what are the ids from ptable vs medium.  Are they potentially identical?  Even so, does it matter?  If you refresh against the ptables association separately from the medium association, then you automatically don't collide, unless I"m missing something.